### ISSUE TITLE
[Question] Scrollabe Form

### ISSUE THREAD

**Issue by LefebvreIlyas** [danvick/flutter_form_builder#523]
Hello !

I need to build a form composed of several fields divided into sections. 
Since there can be many section fields, I use a `ListView.builder`, and each section is a page in a `PageView.builder`.

Here is the structure:

- Form
   - PageView.builder
     - ListView.builder

**Comment by amoslai5128** [comment]
You are right, using `setState` would be expensive when a Formbuilder containing many textfields in a `StatefulWidget`, it might sometime costs you an unexpected rebuild. 
If you can consider using state management method like States_rebuilder, Bloc.

**Comment by LefebvreIlyas** [comment]
Yes that's true, but I just wanted to use FormBuilder.

**Comment by nirbar89** [comment]
@LefebvreIlyas did you solve this? can you share?

**Comment by LefebvreIlyas** [comment]
Sorry for the late reply @nirbar89 .

I use flutter_form_builder to dispatch the data in the different fields BUT I use a cubit to collect the new values of the different fields. And so at each change I rebuild the view.

You can replace the cubit by something else of course, the principle remains the same.

The solution is quite reactive whether on mobile or web.

If my message is not very clear, tell me and I will make a code example to illustrate my solution.

Translated with www.DeepL.com/Translator (free version)

**Comment by nirbar89** [comment]
> Sorry for the late reply @nirbar89.
> 
> I use `flutter_form_builder` to dispatch the data in the different fields BUT I use a `cubit` to collect the new values of the different fields and make the validation. And so at each change I rebuild the view, which corresponds to the idea of @amoslai5128.
> 
> You can replace the `cubit` by something else of course, the principle remains the same.
> 
> The solution is quite reactive whether on mobile or web.
> 
> If my message is not very clear, tell me and I will make a code example to illustrate my solution.
> 
> Have a nice day.

Hey, thank you very much.

I will be more then happy to see an example

**Comment by amoslai5128** [comment]
> > Sorry for the late reply @nirbar89.
> > I use `flutter_form_builder` to dispatch the data in the different fields BUT I use a `cubit` to collect the new values of the different fields and make the validation. And so at each change I rebuild the view, which corresponds to the idea of @amoslai5128.
> > You can replace the `cubit` by something else of course, the principle remains the same.
> > The solution is quite reactive whether on mobile or web.
> > If my message is not very clear, tell me and I will make a code example to illustrate my solution.
> > Have a nice day.
> 
> Hey, thank you very much.
> 
> I will be more then happy to see an example

I've seen a example by using States_rebuilders, every textfeild has a state to rebuild independently and hooks up outside the widget tree. 

Example: https://github.com/GIfatahTH/states_rebuilder/tree/master/examples/ex_002_2_form_validation_with_reactive_model_with_functional_injection

```
On.form(
  () => Column(
    children: <Widget>[
        TextField(
            focusNode: email.focusNode,
            controller: email.controller,
            onSubmitted: (_) {
              password.focusNode.requestFocus();
            },
        ),
        TextField(
            focusNode: password.focusNode,
            controller: password.controller,
            onSubmitted: (_) {
              form.submitFocusNode.requestFocus();
            },
        ),    
    ],
  ),
).listenTo(form),
```

**Comment by deandreamatias** [comment]
This still a issue?

**Comment by deandreamatias** [comment]
Due to lack of response and an old error, I will close this issue.
If the bug still exists, feel free to open a [new issue](https://github.com/danvick/flutter_form_builder/issues/new/choose)
